### PR TITLE
Adding a warning when injected members have public or protected visib…

### PR DESCRIPTION
**Pull request by dlemures**
…ility. solves Issue #119

### CODE REVIEW

File: toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java [hunk 1]
```diff
@@ -153,6 +160,15 @@ private void processInjectAnnotatedMethod(ExecutableElement methodElement,
     methodInjectionTargetList.add(createMethodInjectionTarget(methodElement));
   }
 
+  private void checkAnnotatedMethodNonFatalErrors(ExecutableElement methodElement) {
```
Comment (stephanenicolas) [inline toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java]: I would do that in isValidInjectAnnotatedMethod

File: toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import javax.inject.Inject;
 import javax.lang.model.element.Element;
 import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
```
Comment (stephanenicolas) [inline toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java]: revert this class

File: toothpick-compiler/src/main/java/toothpick/compiler/common/ToothpickProcessor.java [hunk 3]
```diff
@@ -245,6 +247,11 @@ protected boolean isValidInjectAnnotatedMethod(ExecutableElement methodElement)
         return false;
       }
     }
+
+    if (modifiers.contains(PUBLIC) || modifiers.contains(PROTECTED)) {
```
Comment (stephanenicolas) [inline toothpick-compiler/src/main/java/toothpick/compiler/common/ToothpickProcessor.java]: !PACKAGE-PRIVATE is enough as we already test private above

File: toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java (hunk 1 above)
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java]: done

File: toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java (hunk 2 above)
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/memberinjector/MemberInjectorProcessor.java]: done

File: toothpick-compiler/src/main/java/toothpick/compiler/common/ToothpickProcessor.java (hunk 3 above)
Comment (dlemures) [inline toothpick-compiler/src/main/java/toothpick/compiler/common/ToothpickProcessor.java]: not possible 😜

**Comment by stephanenicolas** [comment]
RTM

**Pull request merged**